src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,229 @@
+
+    if (STATUS_SUCCESS == status)
+    {
+        OsConfigLogInfo(m_log, "Telemetry initialized successfully.");
+
+        m_logger = m_logManager->GetLogger(API_KEY, "logger_direct");
+        if (!m_logger)
+        {
+            OsConfigLogError(m_log, "Failed to get logger instance");
+        }
+    }
+    else
+    {
+        OsConfigLogError(m_log, "Telemetry initialization failed. status=%d", status);
+    }
+}
+
+TelemetryManager::~TelemetryManager() noexcept
+{
+    OsConfigLogInfo(m_log, "Telemetry shutting down...");
+    m_logManager->UploadNow();
+    m_logManager.reset();
+    OsConfigLogInfo(m_log, "Telemetry shutdown complete.");
+}
+
+void TelemetryManager::EventWrite(Microsoft::Applications::Events::EventProperties event)
+{
+    m_logger->LogEvent(event);
+}
+
+bool TelemetryManager::ProcessJsonFile(const std::string& filePath)
+{
+    std::ifstream file(filePath);
+    if (!file.is_open())
+    {
+        OsConfigLogError(m_log, "Failed to open file: %s", filePath.c_str());
+        return false;
+    }
+
+    OsConfigLogInfo(m_log, "Processing JSON file: %s", filePath.c_str());
+
+    std::string line;
+    while (std::getline(file, line) && !line.empty())
+    {
+        ProcessJsonLine(line);
+    }
+
+    return true;
+}
+
+bool TelemetryManager::ValidateEventParameters(const std::string& eventName, const std::set<std::string>& jsonKeys)
+{
+    auto it = EVENT_PARAMETER_SETS.find(eventName);
+    if (it == EVENT_PARAMETER_SETS.end())
+    {
+        OsConfigLogError(m_log, "Unknown event type: %s", eventName.c_str());
+        return false;
+    }
+
+    const auto& requiredParams = it->second.first;
+    const auto& optionalParams = it->second.second;
+
+    // Check that all required parameters are present
+    for (const auto& requiredParam : requiredParams)
+    {
+        if (jsonKeys.find(requiredParam) == jsonKeys.end())
+        {
+            OsConfigLogError(m_log, "Missing required parameter '%s' for event '%s'", requiredParam.c_str(), eventName.c_str());
+            return false;
+        }
+    }
+
+    // Check that no unexpected parameters are present
+    for (const auto& jsonKey : jsonKeys)
+    {
+        if (jsonKey == "EventName") continue; // Skip the event name field
+
+        if (requiredParams.find(jsonKey) == requiredParams.end() &&
+            optionalParams.find(jsonKey) == optionalParams.end())
+        {
+            OsConfigLogError(m_log, "Unexpected parameter '%s' for event '%s'", jsonKey.c_str(), eventName.c_str());
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void TelemetryManager::ProcessJsonLine(const std::string& jsonLine)
+{
+    OsConfigLogDebug(m_log, "Processing JSON line: %s", jsonLine.c_str());
+
+    nlohmann::json jsonObject;
+    try
+    {
+        jsonObject = nlohmann::json::parse(jsonLine);
+    }
+    catch(const nlohmann::json::parse_error& e)
+    {
+        // Ignore invalid JSON lines
```

Comment:
log or telemetry here. why would we have invaild lines? Wouldn't we need to investigate and fix?